ray-operator/apis/ray/v1/rayjob_types.go

```diff
@@ -59,7 +59,9 @@ type JobSubmissionMode string
 const (
 	K8sJobMode JobSubmissionMode = "K8sJobMode" // Submit job via Kubernetes Job
 	HTTPMode   JobSubmissionMode = "HTTPMode"   // Submit job via HTTP request
-	UserMode   JobSubmissionMode = "UserMode"   // Don't submit job in KubeRay. Instead, wait for user to submit job and provide the job submission ID
+	// UserMode is deprecated, remove after KubeRay v1.3
+	UserMode        JobSubmissionMode = "UserMode"        // Don't submit job in KubeRay. Instead, wait for user to submit job and provide the job submission ID. Deprecated, use InteractiveMode instead.
```

Comment:
I would prefer to delete `UserMode` directly if we decide to deprecate it, so we don't need to maintain or test its behavior. I think it's fine since it is in alpha status and is only used by the kubectl plugin.

Reply:
+1 for this

Reply:
I'm fine to remove it, I added this here because some users may pick up v1.2.2 and use the API without the plugin, one example was in https://github.com/ray-project/kuberay/pull/2364#issuecomment-2401931746

Reply:
I think removing it is fine, especially since we don't have tests to cover the old behaviors.

Reply:
sounds good, I'll remove this